src/ReachSets/ContinuousPost/BFFPS19/reach_blocks.jl

```diff
@@ -0,0 +1,276 @@
+                       n::Int,
+                       N::Union{Int, Nothing},
+                       output_function::Union{Function, Nothing},
+                       blocks::AbstractVector{Int},
+                       partition::AbstractVector{<:Union{AbstractVector{Int}, Int}},
+                       δ::NUM,
+                       guards_proj::Vector{<:LazySet{NUM}},
+                       block_options,
+                       vars::Vector{Int},
+                       termination::Function,
+                       progress_meter::Union{Progress, Nothing},
+                       res::Vector{<:ReachSet}
+                       )::Tuple{Int, Bool} where {NUM}
+    update!(progress_meter, 1)
+    array = CartesianProductArray(Xhat0[blocks])
+    X_store = (output_function == nothing) ?
+        array :
+        box_approximation(output_function(array))
+    t0 = zero(δ)
+    t1 = δ
+
+    diff_blocks = setdiff(1:length(partition),blocks)
+
+    b = length(blocks)
+    bd = length(diff_blocks)
+
+    array_d = CartesianProductArray(Xhat0[diff_blocks])
+    X_store_d = (output_function == nothing) ?
+        array_d :
+        box_approximation(output_function(array_d))
+
+    terminate, skip, X_store = termination_X(1, X_store, t0, X_store_d, blocks,
+                                diff_blocks, block_options, termination)
+    store!(res, 1, X_store, t0, t1, N)
+    if terminate
+        return 1, skip
+    end
+
+    Xhatk = Vector{LazySet{NUM}}(undef, b)
+    Xhatk_d = Vector{LazySet{NUM}}(undef, bd)
+    ϕpowerk = copy(ϕ)
+    ϕpowerk_cache = similar(ϕ)
+
+    if U != nothing
+        Whatk_blocks = Vector{LazySet{NUM}}(undef, b)
+        Whatk_diff_blocks = Vector{LazySet{NUM}}(undef, bd)
+        inputs = next_set(U)
+        @inbounds for i in 1:b
+            bi = partition[blocks[i]]
+            Whatk_blocks[i] = overapproximate_inputs(1, blocks[i], proj(bi, n) * inputs)
+        end
+        @inbounds for i in 1:bd
+            bi = partition[diff_blocks[i]]
+            Whatk_diff_blocks[i] = overapproximate_inputs(1, diff_blocks[i], proj(bi, n) * inputs)
+        end
+    end
+
+    arr_length = (U == nothing) ? length(partition) : length(partition) + 1
+    arr = Vector{LazySet{NUM}}(undef, arr_length)
+    k = 2
+    @inbounds while true
+        update!(progress_meter, k)
+        X_store = deco_post_dense(b, blocks, Whatk_blocks, partition, ϕpowerk,
+                    arr, arr_length, U, Xhat0, Xhatk, output_function, overapproximate)
+        t0 = t1
+        t1 += δ
+        terminate, skip, reach_set_intersected = termination(k, X_store, t0)
+        if !(isdisjoint(X_store, UnionSetArray(guards_proj)))
+            X_store_d = deco_post_dense(bd, diff_blocks, Whatk_diff_blocks, partition, ϕpowerk, arr,
+                                              arr_length, U, Xhat0, Xhatk_d, output_function, overapproximate)
+
+            X_store = getX_store(reach_set_intersected, X_store_d, block_options, blocks, diff_blocks)
```

Comment:
Here `reach_set_intersected` can be empty.
I first thought about defining `overapproximate(∅::EmptySet, ...) = ∅` in `LazySets`, but then we would store an `EmptySet` as reach set. I think it makes more sense to catch this case and `break`:
```julia
if reach_set_intersected isa EmptySet
    break
end
```
Note that this can only happen here because you take the intersection with the invariant here. The other post operators take the intersection only in the discrete-post phase.